app/models/manageiq/providers/vmware/infra_manager/ems_ref_obj_mixin.rb

```diff
@@ -0,0 +1,7 @@
+module ManageIQ::Providers::Vmware::InfraManager::EmsRefObjMixin
+  extend ActiveSupport::Concern
+
+  def ems_ref_obj
+    @ems_ref_obj ||= VimString.new(ems_ref, ems_ref_type, :ManagedObjectReference)
```

Comment:
You probably also want to clear this value on `ems_ref=`, otherwise you can get yourself into a state where ems_ref does not match ems_ref_obj, but it's memoized and thus wrong.

Reply:
Should this method handle a nil ems_ref as well, if the column hasn't been set yet?  I'd expect it to return nil if ems_ref is nil.

Reply:
:+1: will do

Reply:
Done

Reply:
Where?

Reply:
I was expecting a method here like

```ruby
def ems_ref=(value)
  @ems_ref_obj = nil if value.nil?
  super
end
```

Reply:
okay done, also handling the case where ems_ref is changed without setting ems_ref_obj